packages/host/tests/integration/search-index-test.gts

```diff
@@ -33,3972 +33,4064 @@ const testModuleRealm = 'http://localhost:4202/test/';
+              ],
+            );
+          });
+
+          test('can sort by multiple string field conditions in given directions', async function (assert) {
+            let { data: matching } = await indexer.search({
+              sort: [
+                {
+                  by: 'author.lastName',
+                  on: { module: `${testModuleRealm}book`, name: 'Book' },
+                  direction: 'asc',
+                },
+                {
+                  by: 'author.firstName',
+                  on: { module: `${testModuleRealm}book`, name: 'Book' },
+                  direction: 'desc',
+                },
+              ],
+              filter: {
+                type: { module: `${testModuleRealm}book`, name: 'Book' },
+              },
+            });
+            assert.deepEqual(
+              matching.map((m) => m.id),
+              [
+                `${paths.url}books/2`, // Van Gogh Ab
+                `${paths.url}books/1`, // Mango Ab
+                `${paths.url}books/3`, // Jackie Ag
+                `${paths.url}card-2`, // Cardy --> lastName is null
+              ],
+            );
+          });
+
+          test('can sort by number value', async function (assert) {
+            let { data: matching } = await indexer.search({
+              sort: [
+                {
+                  by: 'editions',
+                  on: { module: `${testModuleRealm}book`, name: 'Book' },
+                },
+                {
+                  by: 'author.lastName',
+                  on: { module: `${testModuleRealm}book`, name: 'Book' },
+                },
+              ],
+              filter: {
+                type: { module: `${testModuleRealm}book`, name: 'Book' },
+              },
+            });
+            assert.deepEqual(
+              matching.map((m) => m.id),
+              [
+                `${paths.url}books/2`, // 0
+                `${paths.url}books/1`, // 1
+                `${paths.url}card-2`, // 1
+                `${paths.url}books/3`, // 2
+              ],
+            );
+          });
+
+          test('can sort by date', async function (assert) {
+            let { data: matching } = await indexer.search({
+              sort: [
+                {
+                  by: 'pubDate',
+                  on: { module: `${testModuleRealm}book`, name: 'Book' },
+                },
+              ],
+              filter: {
+                type: { module: `${testModuleRealm}book`, name: 'Book' },
+              },
+            });
+            // note that sorting by nulls is problematic in that sqlite
+            // considers nulls the smallest possible value and postgres considers
+            // nulls the largest possible value. removing tests that make
+            // assertions around the positions of nulls as it cannot be run
+            // consistently between postgres, sqlite, and our in-memory index
+            assert.deepEqual(
```

Comment:
FYI making sorting assertions around null will be very difficult as there is not a consistent way nulls are treated during sorting. we should just rely on the documented DB behavior here for how nulls are sorted.